.github/workflows/main.yml

```diff
@@ -10,9 +10,10 @@ on:
 name: RUN CODE
 
 env:
-  AR: llvm-ar
   AS: nasm
-  CC: clang
+  CC_x86_64_unknown_uefi: clang
+  AR_x86_64_unknown_uefi: llvm-ar
+  ACTIONS_ALLOW_UNSECURE_COMMANDS: true
```

Comment:
What's this flag used for?

Reply:
`CC_x86_64_unknown_uefi`  supply with certain CC  for x86_64_unkown_uefi target. 
It only impact `x86_64-unknwn-uefi` target build.

| environment | build target | td-shim-tool/ring | td-shim/ring | 
| - | - | - | - |
|CC=clang | host target| clang | NA |
|CC=clang | x86_64-unknown-uefi |  NA  | clang    |
|CC_x86_64_unknown_uefi=clang | host target| host compiler(cl.exe/cc)|  NA |
|CC_x86_64_unknown_uefi=clang | x86_64-unknown-uefi |  NA  | clang    |

FYI:

`cc-rs`: the prioritized order for CC environment variables:

```
<var>_<target> - for example, CC_x86_64-unknown-linux-gnu
<var>_<target_with_underscores> - for example, CC_x86_64_unknown_linux_gnu
<build-kind>_<var> - for example, HOST_CC or TARGET_CFLAGS
<var> - a plain CC, AR as above.
```

Reply:
It would be great to document this, it gives info about how the system works.

Reply:
`clang` is only required when building `ring` with the `x86_64-unknown-uefi` target.  #109